### DISCUSSION TITLE
Restore deleted GitHub commits

### DISCUSSION THREAD

**Discussion by ghost** [community/community#22541] [New to GitHub]
<p>I deleted most of my GitHub commits in a repo. How can I get them back? My local clone is more than a week old…</p>

**Comment by airtower-luna** [comment]
<p>What exactly did you delete? You can’t delete commits via the web interface, unless you delete the whole repository (or branch).</p>

**Comment by ghost** [comment, marked as answer]
<p>whoops! Just realized that it was in my other local clone where I deleted my commits…</p>

**Comment by leilacctcomp** [comment]
I deleted most of my GitHub commits in a repo. How can I get them back? there isn't a local clone

**Comment by jamesdev23** [comment]
To restore a deleted commit in a GitHub repository, you can utilize the git reflog command to locate the commit's SHA hash and then use git reset --hard to restore it. This works by accessing a log of recent actions on the repository, including commits that have been deleted.
